infra/main.bicep

```diff
@@ -0,0 +1,161 @@
+targetScope = 'resourceGroup'
+
+@minLength(1)
+@maxLength(64)
+@description('Name of the the environment which is used to generate a short unique hash used in all resources.')
+param environmentName string
+@minLength(1)
+@description('Primary location for all resources')
+param location string = resourceGroup().location
+param openAiServiceName string = ''
+param openAiSkuName string = 'S0'
+param embeddingDeploymentName string = 'embeddings'
+param embeddingDeploymentCapacity int = 30
+param embeddingModelName string = 'text-embedding-ada-002'
+@description('Id of the user or app to assign application roles')
+param principalId string 
+@secure()
+@description('Application user password')
+param appUserPassword string
+@secure()
+@description('SQL Server administrator password')
+param sqlAdminPassword string
+param appUser string = 'appUser'
+param sqlAdmin string = 'sqlAdmin'
+param dbServiceName string =''
+param keyVaultName string =''
+param dbName string ='sqldb'
+@secure()
+param repositoryToken string
+@secure()
+param repositoryUrl string

```

Comment:
Why respositoryUrl a required string?  Can we figure out a way to make it not required?

Reply:
Modified.